include/fastdds/dds/xtypes/dynamic_types/DynamicData.hpp

```diff
@@ -599,7 +604,7 @@ class RTPS_DllAPI DynamicData : public std::enable_shared_from_this<DynamicData>
      * @retval RETCODE_OK when the value was set successfully.
      * @retval RETCODE_BAD_PARAMETER when the @ref MemberId is invalid or the member type is not promotable to \b complex.
```

Comment:
This value is also returned if the value reference is empty.

Reply:
*std::static_pointer_cast<TypeForKind<TK_STRING8>>(value_iterator->second) = value